core/deployment/src/main/java/io/quarkus/deployment/execannotations/ExecutionModelAnnotationsProcessor.java

```diff
@@ -0,0 +1,94 @@
+package io.quarkus.deployment.execannotations;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.StringJoiner;
+
+import org.jboss.jandex.AnnotationInstance;
+import org.jboss.jandex.AnnotationTarget;
+import org.jboss.jandex.DotName;
+import org.jboss.jandex.IndexView;
+import org.jboss.jandex.MethodInfo;
+import org.jboss.jandex.Type;
+import org.jboss.logging.Logger;
+
+import io.quarkus.deployment.SuppressForbidden;
+import io.quarkus.deployment.annotations.BuildProducer;
+import io.quarkus.deployment.annotations.BuildStep;
+import io.quarkus.deployment.builditem.CombinedIndexBuildItem;
+import io.quarkus.deployment.builditem.GeneratedClassBuildItem;
+import io.smallrye.common.annotation.Blocking;
+import io.smallrye.common.annotation.NonBlocking;
+import io.smallrye.common.annotation.RunOnVirtualThread;
+
+public class ExecutionModelAnnotationsProcessor {
+    private static final Logger log = Logger.getLogger(ExecutionModelAnnotationsProcessor.class);
+
+    private static final DotName BLOCKING = DotName.createSimple(Blocking.class);
+    private static final DotName NON_BLOCKING = DotName.createSimple(NonBlocking.class);
+    private static final DotName RUN_ON_VIRTUAL_THREAD = DotName.createSimple(RunOnVirtualThread.class);
+
+    @BuildStep
+    void check(BuildProducer<GeneratedClassBuildItem> ignored, // only to make sure this build step is executed
```

Comment:
Just FYI our [docs recommend](https://quarkus.io/guides/writing-extensions#artifact-result-build-item) to use `@Produce(ArtifactResultBuildItem.class)` for this special use case but ofc `@Produce(GeneratedClassBuildItem.class)` and equivalents should work as well...

Reply:
Changed.

Reply:
For the record, I need to move back to using `GeneratedClassBuildItem`, because `ArtifactResultBuildItem` apparently isn't consumed in `QuarkusUnitTest`.

Reply:
Interesting. We should probably update the docs then... :thinking: